go/cmd/dolt/commands/reflog.go

```diff
@@ -0,0 +1,206 @@
+	Synopsis: []string{
+		`{{.LessThan}}ref{{.GreaterThan}}`,
+	},
+}
+
+type ReflogCmd struct{}
+
+// Name is returns the name of the Dolt cli command. This is what is used on the command line to invoke the command
+func (cmd ReflogCmd) Name() string {
+	return "reflog"
+}
+
+// Description returns a description of the command
+func (cmd ReflogCmd) Description() string {
+	return "Show history of named refs."
+}
+
+// EventType returns the type of the event to log
+func (cmd ReflogCmd) EventType() eventsapi.ClientEventType {
+	return eventsapi.ClientEventType_REFLOG
+}
+
+func (cmd ReflogCmd) Docs() *cli.CommandDocumentation {
+	ap := cmd.ArgParser()
+	return cli.NewCommandDocumentation(reflogDocs, ap)
+}
+
+func (cmd ReflogCmd) ArgParser() *argparser.ArgParser {
+	return cli.CreateReflogArgParser()
+}
+
+func (cmd ReflogCmd) RequiresRepo() bool {
+	return false
+}
+
+// Exec executes the command
+func (cmd ReflogCmd) Exec(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv, cliCtx cli.CliContext) int {
+	ap := cmd.ArgParser()
+	help, usage := cli.HelpAndUsagePrinters(cli.CommandDocsForCommandString(commandStr, reflogDocs, ap))
+	apr := cli.ParseArgsOrDie(ap, args, help)
+
+	queryist, sqlCtx, closeFunc, err := cliCtx.QueryEngine(ctx)
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	}
+	if closeFunc != nil {
+		defer closeFunc()
+	}
+
+	query, err := constructInterpolatedDoltReflogQuery(apr)
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	}
+
+	rows, err := GetRowsForSql(queryist, sqlCtx, query)
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	}
+
+	return printReflog(rows, queryist, sqlCtx)
+}
+
+// constructInterpolatedDoltReflogQuery generates the sql query necessary to call the DOLT_REFLOG() function.
+// Also interpolates this query to prevent sql injection
+func constructInterpolatedDoltReflogQuery(apr *argparser.ArgParseResults) (string, error) {
+	var params []interface{}
+	refPlaceholder := ""
+	allFlag := ""
+
+	if apr.NArg() == 1 {
+		params = append(params, apr.Arg(0))
+		refPlaceholder = "?"
+	}
+	if apr.Contains(cli.AllFlag) {
+		allFlag = "'--all'"
+	}
+
+	args := ""
+	if refPlaceholder == "" && allFlag != "" {
+		args = allFlag
+	} else if refPlaceholder != "" && allFlag == "" {
+		args = refPlaceholder
+	} else if refPlaceholder != "" && allFlag != "" {
+		args = strings.Join([]string{refPlaceholder, allFlag}, ", ")
+	}
+
+	query := strings.Join([]string{"SELECT ref, commit_hash, commit_message FROM DOLT_REFLOG(", args, ")"}, "")
```

Comment:
(minor) this would be just a _little_ bit simpler as a `Sprintf` call, instead of creating a literal slice and using strings.Join. 

```suggestion
	query := fmt.Sprintf("SELECT ref, commit_hash, commit_message FROM DOLT_REFLOG(%s)"}, args)
```